modules/git/repo_commit.go

```diff
@@ -155,7 +155,7 @@ func (repo *Repository) searchCommits(id SHA1, opts SearchCommitsOptions) ([]*Co
 	if len(opts.Keywords) > 0 {
 		for _, v := range opts.Keywords {
 			// ignore anything below 4 characters as too unspecific
-			if len(v) >= 4 {
+			if len(v) >= 4 && IsValidSHAPattern(v) {
```

Comment:
I think this is a break change and if `IsValidSHAPattern`, we can remove the condition `len(v) >= 4`.

Reply:
no it's not breaking as we did looked for sha IDs before and now we just make sure they realy are